src/ucp/core/ucp_worker.c

```diff
@@ -2948,24 +2951,18 @@ ucp_worker_do_keepalive_progress(ucp_worker_h worker)
         goto out_unblock;
     }
 
-    if (ucs_unlikely(worker->keepalive.iter == &worker->all_eps)) {
-        ucs_assert(worker->keepalive.iter_begin == &worker->all_eps);
-
-        ucp_worker_keepalive_next_ep(worker);
-        worker->keepalive.iter_begin = worker->keepalive.iter;
-    }
-
-    max_ep_count = worker->context->config.ext.keepalive_num_eps;
-    ucs_assert(worker->keepalive.ep_count < max_ep_count);
-
     /* Use own loop for elements because standard for_each skips
      * head element */
     /* TODO: use more optimal algo to enumerate EPs to keepalive
      * (linked list) */
     do {
-        ep = ucp_worker_keepalive_current_ep(worker);
+        ucs_assertv(worker->keepalive.ep_count < max_ep_count,
```

Comment:
why moved inside the loop?

Reply:
moved to the place where `max_ep_count` is initialized